### PR TITLE
WIP: Udayan/newbears

### PR DESCRIPTION
**Pull request by Udayan12167**
And so it begins.

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
thats how you mark it :D

File: coalib/files/Fileproxy.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        """
+        :return: A list of lines of the file to iterate over.
+        """
+        return iter(self.lines)
+
+    def __hash__(self):
+        """
+        :return: Only hashes the filename.
+        """
+        return hash(self.filename)  # pragma: no cover
```
Comment (sils1297) [inline coalib/files/Fileproxy.py]: you can test it by hashing two identical objects and asserting equality

File: coalib/processes/BearRunning.py [hunk 2]
```diff
@@ -408,13 +397,27 @@ def run_local_bears(filename_queue,
     try:
         while True:
             filename = filename_queue.get(timeout=timeout)
+
+            if filename not in file_dict:
+                send_msg(message_queue,
+                         timeout,
+                         LOG_LEVEL.ERROR,
+                         "An internal error occurred.",
+                         Constants.THIS_IS_A_BUG)
```
Comment (sils1297) [inline coalib/processes/BearRunning.py]: can you make a commit removing this whole if? This is an internal error and should never occurs we can just let it throw the exception :)

Comment (Udayan12167) [inline coalib/processes/BearRunning.py]: But who will catch the exception?

Comment (sils1297) [inline coalib/processes/BearRunning.py]: the main routine

**Comment by meetmangukiya** [comment]
@udayan12167 I think @makman2 took this over so close?

**Comment by Makman2** [comment]
Yeah think so. Let's keep the branch around for a while until the nextgen-core is completely implemented.

**Pull request closed without merging**